src/ucp/proto/proto_multi.c

```diff
@@ -84,6 +84,11 @@ ucs_status_t ucp_proto_multi_init(const ucp_proto_multi_init_params_t *params,
     lane_map                = 0;
     max_frag_ratio          = 0;
     for (i = 0; i < num_lanes; ++i) {
+        if (ucs_popcount(lane_map) == params->max_lanes) {
+            ucs_trace("reached max lanes");
```

Comment:
1. add assert that ucs_popcount(lane_map) <= params->max_lanes
2. IMO better remove this ucs_trace

Reply:
Or maybe add `&& ucs_popcount(lane_map) < params->max_lanes` inside the `for` condition instead?

Reply:
instead of assert?

Reply:
@yosefe  Why put an `assert` here? We only need a stop condition for the loop

Reply:
assert that it's always <=
since we stop at ==